### PR TITLE
Make a default selection per network

### PR DESCRIPTION
**Pull request by anxolin**
WIP, TradingWidget has a token selection / network change mess. 
I try to do several workarounds, this is an attempt that apparently works if the user didn't trade DAI before.

### CODE REVIEW

File: src/components/TradeWidget/index.tsx [hunk 1]
```diff
@@ -119,7 +119,8 @@ const validFromId: TradeFormTokenId = 'validFrom'
 const validUntilId: TradeFormTokenId = 'validUntil'
 
 // Grab CONFIG tokens
-const { sellToken: initialSellToken, receiveToken: initialReceiveToken } = CONFIG.initialTokenSelection
+const initialTokenSelection = CONFIG.initialTokenSelection
+const { sellToken: initialSellTokenDefault, receiveToken: initialReceiveTokenDefault } = initialTokenSelection
```
Comment (W3stside) [inline src/components/TradeWidget/index.tsx:123]: why not 1 line this?

File: src/components/TradeWidget/index.tsx [hunk 2]
```diff
@@ -607,7 +591,7 @@ const TradeWidget: React.FC<TradeWidgetProps> = ({
   const onSelectChangeReceiveToken = onSelectChangeFactory(setReceiveToken, sellTokenBalance)
 
   const tokenAddressesToAdd: string[] = useMemo(
-    () => preprocessTokenAddressesToAdd([sellTokenSymbol, receiveTokenSymbol], networkIdOrDefault),
+    () => preprocessTokenAddressesToAdd([sellToken?.symbol, receiveToken.symbol], networkIdOrDefault),
```
Comment (W3stside) [inline src/components/TradeWidget/index.tsx:594]: why is `sellToken?.symbol` but not receiveToken?

File: src/components/TradeWidget/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/components/TradeWidget/index.tsx:123]: yes, sure

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr1518--dexreact.review.gnosisdev.com)**: Testing web app
   * **📚 [Storybook](https://pr1518--dexreact.review.gnosisdev.com/storybook/)**: Component stories

File: src/components/TradeWidget/index.tsx (hunk 2 above)
Comment (anxolin) [reply src/components/TradeWidget/index.tsx:590]: good question, I would swear the two of them were nullables, let me check what i did wrong

File: src/utils/miscellaneous.ts [hunk 3]
```diff
@@ -49,7 +49,11 @@ export function getToken<T extends TokenDetails, K extends keyof T>(
   value: string | number | undefined = '',
   tokens: T[] | undefined | null,
 ): T | undefined {
-  return (tokens || []).find((token: T): boolean => {
+  if (!tokens) {
```
Comment (W3stside) [inline src/utils/miscellaneous.ts:52]: nit: probably cleaner to write it as `if( tokens) { ... })` returns `undefined` if no `tokens`

File: src/components/TradeWidget/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/components/TradeWidget/index.tsx:123]: well, how?

Comment (W3stside) [reply src/components/TradeWidget/index.tsx:123]: here was my implementation when i tried this yesterday:

```ts
// Grab CONFIG tokens
const {
  sellToken: sellTokenConfig,
  receiveToken: receiveTokenConfig,
  networks: networksConfig,
} = CONFIG.initialTokenSelection

function getNetworkSpecificDefaultTokensFromURL(
  networkId: number,
): { initialSellToken: string; initialReceiveToken: string } {
  return {
    initialSellToken: networksConfig[networkId]?.sellToken || sellTokenConfig,
    initialReceiveToken: networksConfig[networkId]?.receiveToken || receiveTokenConfig,
  }
}
```

Comment (anxolin) [reply src/components/TradeWidget/index.tsx:123]: given the times, and that u access the `networksConfig` two times. I think I'll just use mine, although, I think doing a util or placing this function I think somewhere central i think is a good thing. Feel free to improve the RC1

**Pull request merged**